File: BOJ/java/9613.java

import java.util.Scanner;

class Main {
    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while (t-- > 0) {
            int tc = sc.nextInt();
            int[] arr = new int[tc];
            long sum = 0;

            for (int i = 0; i < tc; i++) {
                arr[i] = sc.nextInt();
            }

            for (int i = 0; i < tc; i++) {
                for (int j = i + 1; j < tc; j++) {
                    sum += getGCD(arr[i], arr[j]);
                }
            }
            System.out.println(sum);
        }
        sc.close();
    }

    static int getGCD(int a, int b) {
        while (b != 0) {
            int r = a % b;
            a = b;
            b = r;
        }
        return a;
    }
}

File: BOJ/kotlin/9613.kt
fun main() {
    repeat(readLine()!!.toInt()) {
        val arr = readLine()!!.split(" ").map { it.toLong() }
        val sum = (1 until arr.size).map { i ->
            (1 until i).map { j ->
                getGCD(arr[i], arr[j])
            }.sum()
        }.sum()
        println(sum)
    }

}

fun getGCD(a: Long, b: Long): Long = if (b == 0L) a else getGCD(b, a % b)